### PR TITLE
Adding a test that validates CLI templates don't depend on CLI packages.

### PR DESCRIPTION
**Pull request by livarcocc**
Adding a test that restores all c# templates and checks that none of them depend on a package produced by the CLI. I had to write this as a single test for all templates because it cuts the test time in about half to restore all the packages at once to the same folder than to clean and restore again everytime per template.

Fixes https://github.com/dotnet/cli/issues/4745.

@piotrpMSFT @krwq @jgoshi

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Debian8.2 x64 Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Ubuntu x64 Release Build please.

**Comment by livarcocc** [comment]
@dotnet-bot Test Ubuntu x64 Release Build

**Comment by livarcocc** [comment]
@MattGertz

File: test/dotnet-new.Tests/GivenThatIWantANewCSAppWithSpecifiedType.cs [hunk 1]
```diff
@@ -36,15 +36,56 @@ public class GivenThatIWantANewCSAppWithSpecifiedType : TestBase
                 File.Copy("NuGet.tempaspnetpatch.config", Path.Combine(rootPath, "NuGet.Config"));
             }
 
-            new TestCommand("dotnet") { WorkingDirectory = rootPath }
-                .Execute($"restore /p:SkipInvalidConfigurations=true")
+            new RestoreCommand()
+                .WithWorkingDirectory(rootPath)
+                .Execute("/p:SkipInvalidConfigurations=true")
```
Comment (piotrpMSFT) [inline test/dotnet-new.Tests/GivenThatIWantANewCSAppWithSpecifiedType.cs]: Do we still need this??

File: test/dotnet-new.Tests/GivenThatIWantANewCSAppWithSpecifiedType.cs [hunk 2]
```diff
@@ -36,15 +36,56 @@ public class GivenThatIWantANewCSAppWithSpecifiedType : TestBase
...
-                .ExecuteWithCapturedOutput("build")
+                .ExecuteWithCapturedOutput()
                 .Should().Pass()
                 .And.NotHaveStdErr();
         }
+
+        [Fact]
+        public void RestoreDoesNotUseAnyCliProducedPackagesOnItsTemplates()
+        {
+            var cSharpTemplates = new [] { "Console", "Lib", "Web", "Mstest", "XUnittest" };
```
Comment (piotrpMSFT) [inline test/dotnet-new.Tests/GivenThatIWantANewCSAppWithSpecifiedType.cs]: We need to centralize this list. It's all over the place right now...
